src/iptvsimple/utilities/FileUtils.cpp

```diff
@@ -142,15 +140,19 @@ int FileUtils::GetCachedFileContents(const std::string& cachedName, const std::s
   const std::string cachedPath = FileUtils::GetUserDataAddonFilePath(cachedName);
 
   // check cached file is exists
-  if (useCache && XBMC->FileExists(cachedPath.c_str(), false))
+  if (useCache && kodi::vfs::FileExists(cachedPath, false))
   {
-    struct __stat64 statCached;
-    struct __stat64 statOrig;
+    STAT_STRUCTURE statCached;
+    STAT_STRUCTURE statOrig;
 
-    XBMC->StatFile(cachedPath.c_str(), &statCached);
-    XBMC->StatFile(filePath.c_str(), &statOrig);
+    kodi::vfs::StatFile(cachedPath, statCached);
+    kodi::vfs::StatFile(filePath, statOrig);
 
-    needReload = statCached.st_mtime < statOrig.st_mtime || statOrig.st_mtime == 0;
+#ifdef TARGET_WINDOWS
```

Comment:
Do we really need an ifdef here? Can kodi not do the OS specific stuff internally?

Reply:
Really not nice, thought also about.
I look to make it better, currently is the `stat` on interface to much OS dependent.

Reply:
Does this need to wait for filesystem changes?